src/herder/HerderSCPDriver.cpp

```diff
@@ -1220,7 +1224,27 @@ HerderSCPDriver::checkAndCacheTxSetValid(TxSetFrameConstPtr txSet,
     bool* pRes = mTxSetValidCache.maybeGet(key);
     if (pRes == nullptr)
     {
-        bool res = txSet->checkValid(mApp, closeTimeOffset, closeTimeOffset);
+        bool res = false;
```

Comment:
I admit that I don't fully understand the value of caching here, but if we think it's actually valuable, then we shouldn't re-do the checks, right?

Reply:
My point was to move resolve-related checks out of `checkAndCacheTxSetValid` into `validateValueHelper` to avoid duplicate checks (the code in validateValueHelper already ensures we're at LCL+1) and help with readability

Reply:
This looks more or less like the suggestion after the redesign.